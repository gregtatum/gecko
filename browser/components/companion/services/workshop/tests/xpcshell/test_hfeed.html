
<!-- test components (2 first articles) from CC0-licensed http://microformats.org/wiki/h-feed -->
<div class="h-feed hfeed">
  <h1 class="p-name site-title">The Markup Blog</h1>
  <p class="p-summary site-description">Stories of elements of their attributes.</p>

  <article class="h-entry hentry">
    <a class="u-url" rel="bookmark" href="2020/06/22/balanced-divisive-complementary">
      <h2 class="p-name entry-title">A Tale Of Two Tags: Part 2</h2>
    </a>
    <address class="p-author author h-card vcard">
      <a href="https://chandra.example.com/" class="u-url url p-name fn" rel="author">Chandra</a>
    </address>
    <time class="dt-published published" datetime="2012-06-22T09:45:57-07:00">June 21, 2012</time>
    <div class="p-summary entry-summary">
      <p>From balanced harmony, to divisive misunderstandings, to complementary roles.</p>
    </div>
    <a href="/category/uncategorized/" rel="category tag" class="p-category">General</a>
  </article>

  <article class="h-entry hentry">
    <a class="u-url" rel="bookmark" href="2020/06/20/best-visible-alternative-invisible">
      <h2 class="p-name entry-title">A Tale Of Two Tags: Part 1</h2>
    </a>
    <address class="p-author author h-card vcard">
      <a href="https://chandra.example.com/" class="u-url url p-name fn" rel="author">Chandra</a>
    </address>
    <time class="dt-published published" datetime="2012-06-20T08:34:46-07:00">June 20, 2012</time>
    <div class="p-summary entry-summary">
      <p>It was the best of visible tags, it was the alternative invisible tags.</p>
    </div>
    <a href="/category/uncategorized/" rel="category tag" class="p-category">General</a>
  </article>

  <article class="h-entry hentry">
    <p class="p-name e-content">Hey thanks for making this microformats resource</p>
    <div class="e-content"> <p>Blah blah blah</p> </div>
    <p class="p-name">
      <span>
        <b class="value">M</b>
        <img class="value" alt="o"></img>
        <data class="value" value="z">
          <abbr class="value" title="ignored"></abbr>
        </data>
        <a class="value">i</a>
        <area class="a b c d value e" alt="l"></area>
        <area></area>
        <area>
          <area>
            <area class="value" alt="l"></area>
          </area>
          <abbr class="value">a</abbr>
        </area>
      </span>
    </p>
  </article>

  <article class="h-entry hentry">
    <p> YYYY-MM-DD HH:MM:SS-XX:YY
      <span class="dt-published">
          <time class="value" datetime="1975-03-16">A</time>
       at <img alt="12:34:56-07:08" class="value"></img>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS+XX:YY
      <span class="dt-published">
          <time class="value" datetime="1976-03-16">A</time>
       at <area class="value" alt="12:34:56+07:08"></alt>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS-XXYY
      <span class="dt-published">
          <time class="value" datetime="1977-03-16">A</time>
       at <data class="value" value="12:34:56-0708"></data>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS+XXYY
      <span class="dt-published">
          <time class="value" datetime="1978-03-16">A</time>
       at <abbr class="value" title="12:34:56+0708"></abbr>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SSZ
      <span class="dt-published">
          <time class="value" datetime="1979-03-16">A</time>
       at <del class="value" datetime="12:34:56Z"></del>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS -XX:YY
      <span class="dt-published">
          <time class="value" datetime="1980-03-16">A</time>
       at <ins class="value" datetime="12:34:56"></ins>
          <time class="value">-07:08</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS +XX:YY
      <span class="dt-published">
          <time class="value" datetime="1981-03-16">A</time>
       at <time class="value">12:34:56</time>
          <time class="value">+07:08</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS -XXYY
      <span class="dt-published">
          <time class="value" datetime="1982-03-16">A</time>
       at <time class="value">12:34:56</time>
          <time class="value">-0708</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS +XXYY
      <span class="dt-published">
          <time class="value" datetime="1983-03-16">A</time>
       at <time class="value">
         12:34:56
        </time>
          <time class="value">

            +0708

          </time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS -XX
      <span class="dt-published">
          <time class="value" datetime="1984-03-16">A</time>
       at <time class="value">12:34:56</time>
          <time class="value">-07</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS +XX
      <span class="dt-published">
          <time class="value" datetime="1985-03-16">A</time>
       at <time class="value">12:34:56</time>
          <time class="value">+07</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM:SS Z
      <span class="dt-published">
          <time class="value" datetime="1986-03-16">A</time>
       at <time class="value">12:34:56</time>
          <time class="value">Z</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM-XX:YY
      <span class="dt-published">
          <time class="value" datetime="1987-03-16">A</time>
       at <time class="value">12:34-07:08</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM+XX:YY
      <span class="dt-published">
          <time class="value" datetime="1988-03-16">A</time>
       at <time class="value">12:34+07:08</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM-XXYY
      <span class="dt-published">
          <time class="value" datetime="1989-03-16">A</time>
       at <time class="value">12:34-0708</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM+XXYY
      <span class="dt-published">
          <time class="value" datetime="1990-03-16">A</time>
       at <time class="value">12:34+0708</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MMZ
      <span class="dt-published">
          <time class="value" datetime="1991-03-16">A</time>
       at <time class="value">12:34Z</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM
      <span class="dt-published">
          <time class="value" datetime="1992-03-16">A</time>
       at <time class="value">12:34</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM (the second time must be ignored)
      <span class="dt-published">
          <time class="value" datetime="1993-03-16">A</time>
       at <time class="value">12:34</time>
       and at <time class="value">01:02</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM (the value "foo bar" must be ignored)
      <span class="dt-published">
          <time class="value" datetime="1994-03-16">A</time>
       at <time class="value">foo bar</time>
       and at <time class="value">12:34</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MM (the time must be ignored because it's nested in a value)
      <span class="dt-published">
          <time class="value" datetime="1995-03-16">A</time>
       at <data class="value" value="foo bar">
        and at <time class="value">12:34</time>
       </data>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MMam
      <span class="dt-published">
          <time class="value" datetime="1996-03-16">A</time>
       at <time class="value">01:23am</time>
      </span>
    </p>

    <p> YYYY-MM-DD HH:MMa.m.
      <span class="dt-published">
          <time class="value" datetime="1997-03-16">A</time>
       at <time class="value">01:23a.m.</time>
      </span>
    </p>

  <p> YYYY-MM-DD HH:MMpm
    <span class="dt-published">
        <time class="value" datetime="1998-03-16">A</time>
     at <time class="value">01:23pm</time>
    </span>
  </p>

  <p> YYYY-MM-DD HH:MMp.m.
    <span class="dt-published">
        <time class="value" datetime="1999-03-16">A</time>
     at <time class="value">01:23p.m.</time>
    </span>
  </p>

  <p> YYYY-MM-DD HH:MM:SSam
    <span class="dt-published">
        <time class="value" datetime="2000-03-16">A</time>
     at <time class="value">01:23:45am</time>
    </span>
  </p>

  <p> YYYY-MM-DD HH:MM:SSa.m.
    <span class="dt-published">
        <time class="value" datetime="2001-03-16">A</time>
     at <time class="value">01:23:45a.m.</time>
    </span>
  </p>

<p> YYYY-MM-DD HH:MM:SSpm
  <span class="dt-published">
      <time class="value" datetime="2002-03-16">A</time>
   at <time class="value">01:23:45pm</time>
  </span>
</p>

<p> YYYY-MM-DD HH:MM:SSp.m.
  <span class="dt-published">
      <time class="value" datetime="2003-03-16">A</time>
   at <time class="value">01:23:45p.m.</time>
  </span>
</p>
</article>

</div>
