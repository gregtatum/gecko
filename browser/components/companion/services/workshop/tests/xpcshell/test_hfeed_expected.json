{
  "entries": [
    {
      "url": [
        "https://www.allizom.org/2020/06/22/balanced-divisive-complementary",
        "https://chandra.example.com/"
      ],
      "name": ["A Tale Of Two Tags: Part 2", "Chandra"],
      "author": ["Chandra"],
      "published": ["2012-06-22T16:45:57.000Z"],
      "summary": [
        "From balanced harmony, to divisive misunderstandings, to complementary roles."
      ],
      "category": ["General"]
    },
    {
      "url": [
        "https://www.allizom.org/2020/06/20/best-visible-alternative-invisible",
        "https://chandra.example.com/"
      ],
      "name": ["A Tale Of Two Tags: Part 1", "Chandra"],
      "author": ["Chandra"],
      "published": ["2012-06-20T15:34:46.000Z"],
      "summary": [
        "It was the best of visible tags, it was the alternative invisible tags."
      ],
      "category": ["General"]
    },
    {
      "name": ["Hey thanks for making this microformats resource", "Mozilla"],
      "content": [
        {
          "value": "Hey thanks for making this microformats resource",
          "html": "<html><head></head><body>Hey thanks for making this microformats resource</body></html>"
        },
        {
          "value": "Blah blah blah",
          "html": "<html><head></head><body><p>Blah blah blah</p> </body></html>"
        }
      ]
    },
    {
      "published": [
        "1975-03-16T19:42:56.000Z",
        "1976-03-16T05:26:56.000Z",
        "1977-03-16T19:42:56.000Z",
        "1978-03-16T05:26:56.000Z",
        "1979-03-16T12:34:56.000Z",
        "1980-03-16T19:42:56.000Z",
        "1981-03-16T05:26:56.000Z",
        "1982-03-16T19:42:56.000Z",
        "1983-03-16T05:26:56.000Z",
        "1984-03-16T19:34:56.000Z",
        "1985-03-16T05:34:56.000Z",
        "1986-03-16T12:34:56.000Z",
        "1987-03-16T19:42:00.000Z",
        "1988-03-16T05:26:00.000Z",
        "1989-03-16T19:42:00.000Z",
        "1990-03-16T05:26:00.000Z",
        "1991-03-16T12:34:00.000Z",
        "1992-03-16T12:34:00.000Z",
        "1993-03-16T12:34:00.000Z",
        "1994-03-16T12:34:00.000Z",
        "1995-03-16T00:00:00.000Z",
        "1996-03-16T01:23:00.000Z",
        "1997-03-16T01:23:00.000Z",
        "1998-03-16T13:23:00.000Z",
        "1999-03-16T13:23:00.000Z",
        "2000-03-16T01:23:45.000Z",
        "2001-03-16T01:23:45.000Z",
        "2002-03-16T13:23:45.000Z",
        "2003-03-16T13:23:45.000Z"
      ]
    }
  ],
  "name": ["The Markup Blog"]
}
